program.py

```diff
@@ -181,990 +156,204 @@ def send_omf_message_to_endpoint(message_type, message_omf_json, action='create'
+    # Step 1 - Read endpoint configurations from config.json
+    endpoints = get_config()
+
+    # Step 2 - Get OMF Types
+    omf_types = get_json_file("OMF-Types.json")
+
+    # Step 3 - Get OMF Containers
+    omf_containers = get_json_file("OMF-Containers.json")
+
+    # Step 4 - Get OMF Data
+    omf_data = get_json_file("OMF-Data.json")
 
-        # Step 2
-        getToken()
+    # Send messages and check for each endpoint in config.json
 
-        # Steps 3-8 contained in here
-        oneTimeSendMessages()
+    try:
+        # Send out the messages that only need to be sent once
+        for endpoint in endpoints:
+            if not endpoint["verify-ssl"]:
+                print("You are not verifying the certificate of the end point.  This is not advised for any system as there are security issues with doing this.")
+
+            # Step 5 - Send OMF Types
+            for omf_type in omf_types:
+                send_message_to_omf_endpoint(endpoint, "type", [omf_type])
+
+            # Step 6 - Send OMF Containers
+            for omf_container in omf_containers:
+                send_message_to_omf_endpoint(
+                    endpoint, "container", [omf_container])
 
-        # Step 9
+        # Step 7 - Send OMF Data
         count = 0
-        lastVal = ''
-        time.sleep(1)
-        while count == 0 or ((not test) and count < 2):
-            val = create_data_values_for_first_dynamic_type("Container1")
-            lastVal = val
-            send_omf_message_to_endpoint("data", val)
-            send_omf_message_to_endpoint(
-                "data", create_data_values_for_first_dynamic_type("Container2"))
-            send_omf_message_to_endpoint(
-                "data", create_data_values_for_second_dynamic_type("Container3"))
-            send_omf_message_to_endpoint(
-                "data", create_data_values_for_third_dynamic_type("Container4"))
-            if(sendingToOCS):
-                send_omf_message_to_endpoint(
-                    "data", create_data_values_for_NonTimeStampIndexAndMultiIndex_type("Container5", "Container6"))
+        # send data to all endpoints forever if this is not a test
+        while not test or count < 2:
+
+            '''This is where custom loop logic should go. 
+            The get_data call should also be customized to populate omf_data with relevant data.'''
+
+            for omf_datum in omf_data:
+                data_to_send = get_data(omf_datum)
+                for endpoint in endpoints:
+                    # send the data
+                    send_message_to_omf_endpoint(
+                        endpoint, "data", [data_to_send])
+                    
+                # record the values sent if this is a test
+                if test and count == 1:
+                    last_sent_values.update({omf_datum["containerid"] : data_to_send})
+
             time.sleep(1)
```

Comment:
Why the sleep?

Reply:
Sometimes I get tired

Reply:
I added a constant to make this configurable